gradle/tapiTest.gradle

```diff
@@ -0,0 +1,110 @@
+import org.gradle.testing.IntegrationTest
+
+apply plugin: 'java'
+
+sourceSets {
+    tapiTest {
+        compileClasspath += main.output
+        runtimeClasspath += main.output
+    }
+}
+
+configurations {
+    tapiTestCompile.extendsFrom testCompile
+    tapiTestImplementation.extendsFrom testImplementation
+    tapiTestRuntime.extendsFrom testRuntime
+}
+
+dependencies {
+    tapiTestCompile project(":internalIntegTesting")
+
+    //so that implicit help tasks are available:
+    tapiTestRuntime project(':diagnostics')
+
+    //So that the wrapper and init task are added when tapiTests are run via commandline
+    tapiTestRuntime project(':buildInit')
+    //above can be removed when we implement the auto-apply plugins
+
+    tapiTestRuntime project(":toolingApiBuilders")
+}
+
+integTestTasks.all { IntegrationTest task ->
+    if(task.name == "tapiTest" || task.name.endsWith("TapiTest")) {
+        testClassesDirs = sourceSets.tapiTest.output.classesDirs
+        classpath = sourceSets.tapiTest.runtimeClasspath
+        requiresBinZip = true
+        requiresLibsRepo = true
+    }
+}
+
+task tapiTest(type: IntegrationTest) {
+    def defaultExecuter = project.hasProperty("defaultIntegTestExecuter") ? project.defaultIntegTestExecuter : "embedded"
+    description "Runs tapiTests with '${defaultExecuter}' executer"
+    systemProperties['org.gradle.integtest.executer'] = defaultExecuter
+    if (project.hasProperty('org.gradle.integtest.debug')) {
+        systemProperties['org.gradle.integtest.debug'] = 'true'
+        testLogging.showStandardStreams = true
+    }
+    if (project.hasProperty('org.gradle.integtest.verbose')) {
+        testLogging.showStandardStreams = true
+    }
+    if(project.hasProperty('org.gradle.integtest.launcher.debug')) {
+        systemProperties['org.gradle.integtest.launcher.debug'] = 'true'
+    }
+}
+check.dependsOn(tapiTest)
+
+['embedded', 'forking', 'noDaemon', 'parallel'].each { mode ->
```

Comment:
This doesn't make sense for TAPI tests, they are either embedded or forking. They don't support 'parallel' or 'noDaemon'.

We should only have the normal "tapiTest" above and the "gradleXXXTapiTest"s below.